### PR TITLE
[SPARK-17208] [SQL] Build Outer Join Test Cases in File-based Testing Framework

### PR DESCRIPTION
**Pull request by gatorsmile**
### What changes were proposed in this pull request?

This PR is part of the original PR: https://github.com/apache/spark/pull/14625.

Based on file-based SQL end-to-end testing framework in `SQLQueryTestSuite`, this PR is to create test cases for outer joins. 

In addition, this PR also renamed the test data sets to more meaningful names, based on the comments from @cloud-fan 
### How was this patch tested?

Using the same data and query, the Hive outputs are attached below:
[hive.output.txt](https://github.com/apache/spark/files/434258/hive.output.txt)

After comparison, the Spark-generated results are the same.

### CODE REVIEW

File: sql/core/src/test/scala/org/apache/spark/sql/SQLQueryTestSuite.scala [hunk 1]
```diff
@@ -243,19 +243,81 @@ class SQLQueryTestSuite extends QueryTest with SharedSQLContext {
...
+    // A data set containing a complex data type: MAP
     (Tuple1(Map(1 -> "a1", 2 -> "b1", 3 -> "c1", 4 -> "d1", 5 -> "e1")) ::
       Tuple1(Map(1 -> "a2", 2 -> "b2", 3 -> "c2", 4 -> "d2")) ::
       Tuple1(Map(1 -> "a3", 2 -> "b3", 3 -> "c3")) ::
       Tuple1(Map(1 -> "a4", 2 -> "b4")) ::
       Tuple1(Map(1 -> "a5")) :: Nil)
       .toDF("mapcol")
       .createOrReplaceTempView("mapdata")
+
+    // A data set containing uppercase column names and column values
```
Comment (rxin) [inline sql/core/src/test/scala/org/apache/spark/sql/SQLQueryTestSuite.scala]: do we really need to add these tables here? can we just use temp views on the test cases?

Comment (gatorsmile) [inline sql/core/src/test/scala/org/apache/spark/sql/SQLQueryTestSuite.scala]: Sure, will move them into the test cases.

**Comment by gatorsmile** [comment]
Using the same data and query, the Hive outputs are attached below:
[hive.output.txt](https://github.com/apache/spark/files/434258/hive.output.txt)

After comparison, the Spark-generated results are the same.

**Comment by rxin** [comment]
@gatorsmile maybe you should stop porting more tests for now. I don't want you to waste time. I feel you are just porting the tests over from Hive compatibility suite, but that's not really the point. This is a framework to create test cases that would actually make sense. The best cases are the ones that are very small and simple and can test specific aspects of the query engine.

**Comment by gatorsmile** [comment]
I see. Thanks! Let me close this one.

**Pull request closed without merging**